broken-site/broken-site-impl/src/main/java/com/duckduckgo/brokensite/impl/RealBrokenSitePrompt.kt

```diff
@@ -88,15 +96,48 @@ class RealBrokenSitePrompt @Inject constructor(
     }
 
     override suspend fun shouldShowBrokenSitePrompt(url: String): Boolean {
-        return isFeatureEnabled() &&
-            getUserRefreshesCount() >= REFRESH_COUNT_LIMIT &&
-            brokenSiteReportRepository.getNextShownDate()?.isBefore(currentTimeProvider.localDateTimeNow()) ?: true &&
-            !duckGoUrlDetector.isDuckDuckGoUrl(url)
+        if (!isFeatureEnabled() || getUserRefreshesCount() < REFRESH_COUNT_LIMIT || duckGoUrlDetector.isDuckDuckGoUrl(url)) {
+            return false
+        }
+
+        val currentTimestamp = currentTimeProvider.localDateTimeNow()
+
+        brokenSiteReportRepository.getNextShownDate()?.let { nextDate ->
+            if (currentTimestamp.isBefore(nextDate)) {
+                return false
+            }
+        }
+
+        val dismissStreakResetDays = brokenSiteReportRepository.getDismissStreakResetDays().toLong()
+        when (val lastShownDate = brokenSiteReportRepository.getLastShownDate()) {
+            null -> {
+                // First time showing the prompt
+                return true
+            }
+
+            else -> {
+                if (currentTimestamp.isAfter(lastShownDate.plusDays(dismissStreakResetDays))) {
```

Comment:
We shouldn't check the current timestamp against `dismissStreakResetDays` here, but rather `getDismissalCountBetween`. If the number is lower than `getMaxDismissStreak`, at this point we should return true (since we've already checked `getNextShownDate` above)